fri/src/verifier/mod.rs

```diff
@@ -378,3 +346,14 @@ fn get_query_values<E: FieldElement, const N: usize>(
 
     result
 }
+
+// Evaluates a polynomial with coefficients in an extension field at a point in the base field.
+pub fn eval_horner<E, B>(p: &[E], x: B) -> E
+where
+    E: FieldElement<BaseField = B>,
+    B: FieldElement,
```

Comment:
Couldn't we abstract the second type and refer to `E::BaseField`? Something like:

```suggestion
pub fn eval_horner<E>(p: &[E], x: E::BaseField) -> E
where
    E: FieldElement,
```